swarmrl/networks/flax_network.py

```diff
@@ -64,7 +65,8 @@ def __init__(
             rng_key = onp.random.randint(0, 1027465782564)
         self.sampling_strategy = sampling_strategy
         self.model = flax_model
-        self.apply_fn = jax.jit(self.model.apply)
+        self.apply_fn = jax.jit(jax.vmap(self.model.apply, in_axes=(None, 0)))
```

Comment:
for minor speedup:

```python
self.batch_fn = jax.vmap(self.model.apply, in_axes=(None, 0))
self.apply_fn = jax.jit(self.batch_fn)
```

Reply:
Not sure about that. 
model.apply computes logits for single observable 
apply_fn computes logits for all colloids (array of observables)
batch_apply computes logits for an whole episode (array of array of observables)

Reply:
mm I see the point, there is some inherent mapping over batch axes in Flax which I think would take care of that. Can we test it.